src/common/datatypes/Map.h

```diff
@@ -22,6 +24,22 @@ struct Map {
     kvs = std::move(values);
   }
 
+  // constructor to convert from json string
+  explicit Map(const std::string& json) {
+    auto obj = folly::parseJson(json);
+    for (auto& kv : obj.items()) {
+      kvs.emplace(kv.first.getString(), Value(kv.second));
+    }
+  }
+
+  // constructor to convert from Value::Type::STRING in JSON format
+  explicit Map(const Value& value) {
```

Comment:
Add a Dcheck to check the value type is str, or just delete this interface.

Reply:
Got it, done by removing it :)